pkg/authz/response.go

```diff
@@ -70,144 +95,341 @@ func (d *AuthzData) FilterResp(resp *http.Response) error {
 		return err
 	}
 
-	pom := partialObjectOrList{}
-	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(body), 100).Decode(&pom); err != nil {
-		fmt.Println(err)
-		return err
+	// Use proper Kubernetes content negotiation for stream decoding
+	contentType := resp.Header.Get("Content-Type")
+	mediaType, params, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		return fmt.Errorf("failed to parse content type %s: %v", contentType, err)
 	}
-	klog.V(3).InfoS("upstream response object", "name", pom.Name, "namespace", pom.Namespace, "kind", pom.Kind, "item_count", len(pom.Items))
-	klog.V(4).InfoS("upstream response object detail", "pom", pom)
 
-	var filtered []byte
-	switch {
-	case pom.GroupVersionKind().GroupKind() == schema.GroupKind{Group: "meta.k8s.io", Kind: "Table"}:
-		filtered, err = d.FilterTable(body)
-	case pom.IsList():
-		filtered, err = d.FilterList(body)
-	default:
-		// filterResponse single object
-		filtered, err = d.FilterObject(pom.ToPartialObjectMetadata(), body)
+	var filteredBody bytes.Buffer
+	var filterErr error
+
+	// detect if it's a table request
+	// TODO: see if there's a helper for this
+	if strings.Contains(resp.Request.Header.Get("Accept"), "as=Table") {
+		// TODO: match the other new filter apis, pass object
```

Comment:
Same here

Reply:
https://github.com/authzed/spicedb-kubeapi-proxy/issues/117 (bundling that in here)